### ISSUE TITLE
`MOODLE_SKIP_INSTALL` broken since 3.10.0-debian-10-r45

### ISSUE THREAD

**Issue by vlk-charles** [bitnami/bitnami-docker-moodle#166]
**Description**

Since 3.10.0-debian-10-r45, running a new container with `-e MOODLE_SKIP_INSTALL=yes` against a prepopulated database crashes it. This worked in 3.9 and also 3.10.0-debian-10-r44.

**Steps to reproduce the issue:**

1. Create a new `mariadb` instance with an empty `bitnami_moodle` database.
2. Run a container based on `bitnami/moodle:3.10.1-debian-10-r0` and let it populate the DB from the previous container.
3. Stop the previous Moodle container after it is done initializing.
4. Run a new Moodle container with `MOODLE_SKIP_INSTALL=yes` set.

```
# docker create network moodle-test
# docker run --rm -d --name moodle-test-db --network moodle-test -e MYSQL_ALLOW_EMPTY_PASSWORD=1 mariadb
# docker exec moodle-test-db mysql -e 'create database bitnami_moodle;'
# docker run --rm --network moodle-test -e MARIADB_HOST=moodle-test-db -e MOODLE_DATABASE_USER=root -e ALLOW_EMPTY_PASSWORD=yes bitnami/moodle:3.10.1-debian-10-r0
moodle 22:09:56.27 
moodle 22:09:56.28 Welcome to the Bitnami moodle container
moodle 22:09:56.28 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-moodle
moodle 22:09:56.28 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-moodle/issues
moodle 22:09:56.28 
moodle 22:09:56.28 INFO  ==> ** Starting Moodle setup **
moodle 22:09:56.45 INFO  ==> Configuring PHP options
moodle 22:09:56.48 INFO  ==> Validating settings in MYSQL_CLIENT_* env vars
moodle 22:09:56.53 WARN  ==> You set the environment variable ALLOW_EMPTY_PASSWORD=yes. For safety reasons, do not use this flag in a production environment.
moodle 22:09:56.60 INFO  ==> Ensuring Moodle directories exist
moodle 22:09:56.63 INFO  ==> Trying to connect to the database server
moodle 22:09:56.66 INFO  ==> Running Moodle install script
moodle 22:16:54.32 INFO  ==> Persisting Moodle installation
moodle 22:17:25.70 INFO  ==> ** Moodle setup finished! **

moodle 22:17:25.72 INFO  ==> ** Starting cron **
moodle 22:17:26.16 INFO  ==> ** Starting Apache **
[Tue Jan 19 22:17:26.240230 2021] [ssl:warn] [pid 1] AH01909: www.example.com:8443:0 server certificate does NOT include an ID which matches the server name
[Tue Jan 19 22:17:26.241006 2021] [ssl:warn] [pid 1] AH01909: www.example.com:8443:0 server certificate does NOT include an ID which matches the server name
[Tue Jan 19 22:17:26.296141 2021] [ssl:warn] [pid 1] AH01909: www.example.com:8443:0 server certificate does NOT include an ID which matches the server name
[Tue Jan 19 22:17:26.296843 2021] [ssl:warn] [pid 1] AH01909: www.example.com:8443:0 server certificate does NOT include an ID which matches the server name
[Tue Jan 19 22:17:26.325830 2021] [mpm_prefork:notice] [pid 1] AH00163: Apache/2.4.46 (Unix) OpenSSL/1.1.1d PHP/7.3.26 configured -- resuming normal operations
[Tue Jan 19 22:17:26.325898 2021] [core:notice] [pid 1] AH00094: Command line: '/opt/bitnami/apache/bin/httpd -f /opt/bitnami/apache/conf/httpd.conf -D FOREGROUND'
^C[Tue Jan 19 22:19:32.381302 2021] [mpm_prefork:notice] [pid 1] AH00169: caught SIGTERM, shutting down
# echo $?
0
# docker run --rm --network moodle-test -e MARIADB_HOST=moodle-test-db -e MOODLE_DATABASE_USER=root -e ALLOW_EMPTY_PASSWORD=yes -e MOODLE_SKIP_INSTALL=yes bitnami/moodle:3.10.1-debian-10-r0
moodle 22:21:51.24 
moodle 22:21:51.24 Welcome to the Bitnami moodle container
moodle 22:21:51.24 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-moodle
moodle 22:21:51.24 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-moodle/issues
moodle 22:21:51.24 
moodle 22:21:51.25 INFO  ==> ** Starting Moodle setup **
moodle 22:21:51.41 INFO  ==> Configuring PHP options
moodle 22:21:51.43 INFO  ==> Validating settings in MYSQL_CLIENT_* env vars
moodle 22:21:51.49 WARN  ==> You set the environment variable ALLOW_EMPTY_PASSWORD=yes. For safety reasons, do not use this flag in a production environment.
moodle 22:21:51.56 INFO  ==> Ensuring Moodle directories exist
moodle 22:21:51.59 INFO  ==> Trying to connect to the database server
moodle 22:21:51.61 INFO  ==> An already initialized Moodle database was provided, it will not be re-initialized
moodle 22:21:51.62 INFO  ==> Creating Moodle configuration file
moodle 22:21:53.98 INFO  ==> Running database upgrade
# echo $?
127
```

**Describe the results you received:**

As can be seen from the previous output, the container does not keep running and instead exits after "Running database upgrade" with status 127. This is consistently reproducible.

**Describe the results you expected:**

When the same steps are performed on `3.9.2-debian-10-r47`, the container keeps running until I terminate it manually.
```
# docker exec moodle-test-db mysql -e 'drop database bitnami_moodle; create database bitnami_moodle;'
# docker run --rm --network moodle-test -e MARIADB_HOST=moodle-test-db -e MOODLE_DATABASE_USER=root -e ALLOW_EMPTY_PASSWORD=yes bitnami/moodle:3.9.2-debian-10-r47
moodle 22:25:04.56 
moodle 22:25:04.56 Welcome to the Bitnami moodle container
moodle 22:25:04.57 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-moodle
moodle 22:25:04.57 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-moodle/issues
moodle 22:25:04.57 
moodle 22:25:04.58 INFO  ==> ** Starting Moodle setup **
moodle 22:25:04.74 INFO  ==> Configuring PHP options
moodle 22:25:04.74 INFO  ==> Setting PHP expose_php option
moodle 22:25:04.78 INFO  ==> Validating settings in MYSQL_CLIENT_* env vars
moodle 22:25:04.83 WARN  ==> You set the environment variable ALLOW_EMPTY_PASSWORD=yes. For safety reasons, do not use this flag in a production environment.
moodle 22:25:04.91 INFO  ==> Ensuring Moodle directories exist
moodle 22:25:04.94 INFO  ==> Trying to connect to the database server
moodle 22:25:04.96 INFO  ==> Running Moodle install script
moodle 22:30:47.56 INFO  ==> Persisting Moodle installation
moodle 22:30:59.55 INFO  ==> ** Moodle setup finished! **

moodle 22:30:59.56 INFO  ==> ** Starting cron **
moodle 22:30:59.60 INFO  ==> ** Starting Apache **
[Tue Jan 19 22:30:59.684945 2021] [ssl:warn] [pid 1] AH01909: www.example.com:8443:0 server certificate does NOT include an ID which matches the server name
[Tue Jan 19 22:30:59.685791 2021] [ssl:warn] [pid 1] AH01909: www.example.com:8443:0 server certificate does NOT include an ID which matches the server name
[Tue Jan 19 22:30:59.745043 2021] [ssl:warn] [pid 1] AH01909: www.example.com:8443:0 server certificate does NOT include an ID which matches the server name
[Tue Jan 19 22:30:59.746102 2021] [ssl:warn] [pid 1] AH01909: www.example.com:8443:0 server certificate does NOT include an ID which matches the server name
[Tue Jan 19 22:30:59.775491 2021] [mpm_prefork:notice] [pid 1] AH00163: Apache/2.4.46 (Unix) OpenSSL/1.1.1d PHP/7.3.24 configured -- resuming normal operations
[Tue Jan 19 22:30:59.775554 2021] [core:notice] [pid 1] AH00094: Command line: '/opt/bitnami/apache/bin/httpd -f /opt/bitnami/apache/conf/httpd.conf -D FOREGROUND'
^C[Tue Jan 19 22:31:57.440961 2021] [mpm_prefork:notice] [pid 1] AH00169: caught SIGTERM, shutting down
# echo $?
0
# docker run --rm --network moodle-test -e MARIADB_HOST=moodle-test-db -e MOODLE_DATABASE_USER=root -e ALLOW_EMPTY_PASSWORD=yes -e MOODLE_SKIP_INSTALL=yes bitnami/moodle:3.9.2-debian-10-r47
moodle 22:33:05.88 
moodle 22:33:05.88 Welcome to the Bitnami moodle container
moodle 22:33:05.88 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-moodle
moodle 22:33:05.88 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-moodle/issues
moodle 22:33:05.89 
moodle 22:33:05.89 INFO  ==> ** Starting Moodle setup **
moodle 22:33:06.16 INFO  ==> Configuring PHP options
moodle 22:33:06.16 INFO  ==> Setting PHP expose_php option
moodle 22:33:06.20 INFO  ==> Validating settings in MYSQL_CLIENT_* env vars
moodle 22:33:06.25 WARN  ==> You set the environment variable ALLOW_EMPTY_PASSWORD=yes. For safety reasons, do not use this flag in a production environment.
moodle 22:33:06.32 INFO  ==> Ensuring Moodle directories exist
moodle 22:33:06.35 INFO  ==> Trying to connect to the database server
moodle 22:33:06.37 INFO  ==> An already initialized Moodle database was provided, it will not be re-initialized
moodle 22:33:06.38 INFO  ==> Creating Moodle configuration file
moodle 22:33:08.66 INFO  ==> Running database upgrade
moodle 22:33:09.62 INFO  ==> Persisting Moodle installation
moodle 22:33:21.70 INFO  ==> ** Moodle setup finished! **

moodle 22:33:21.72 INFO  ==> ** Starting cron **
moodle 22:33:21.76 INFO  ==> ** Starting Apache **
[Tue Jan 19 22:33:21.844246 2021] [ssl:warn] [pid 1] AH01909: www.example.com:8443:0 server certificate does NOT include an ID which matches the server name
[Tue Jan 19 22:33:21.845033 2021] [ssl:warn] [pid 1] AH01909: www.example.com:8443:0 server certificate does NOT include an ID which matches the server name
[Tue Jan 19 22:33:21.899853 2021] [ssl:warn] [pid 1] AH01909: www.example.com:8443:0 server certificate does NOT include an ID which matches the server name
[Tue Jan 19 22:33:21.900538 2021] [ssl:warn] [pid 1] AH01909: www.example.com:8443:0 server certificate does NOT include an ID which matches the server name
[Tue Jan 19 22:33:21.929680 2021] [mpm_prefork:notice] [pid 1] AH00163: Apache/2.4.46 (Unix) OpenSSL/1.1.1d PHP/7.3.24 configured -- resuming normal operations
[Tue Jan 19 22:33:21.929745 2021] [core:notice] [pid 1] AH00094: Command line: '/opt/bitnami/apache/bin/httpd -f /opt/bitnami/apache/conf/httpd.conf -D FOREGROUND'
^C[Tue Jan 19 22:34:55.210624 2021] [mpm_prefork:notice] [pid 1] AH00169: caught SIGTERM, shutting down
# echo $?
0
```

**Version**

- Output of `docker version`:

```
Client: Docker Engine - Community
 Version:           19.03.5
 API version:       1.40
 Go version:        go1.12.12
 Git commit:        633a0ea838
 Built:             Wed Nov 13 07:26:10 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.5
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.12.12
  Git commit:       633a0ea838
  Built:            Wed Nov 13 07:24:42 2019
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.2.10
  GitCommit:        b34a5c8af56e510852c35414db4c1f4fa6172339
 runc:
  Version:          1.0.0-rc8+dev
  GitCommit:        3e425f80a8c931f88e6d94a8c831b9d5aa481657
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```

- Output of `docker info`:

```
Client:
 Debug Mode: false

Server:
 Containers: 11
  Running: 11
  Paused: 0
  Stopped: 0
 Images: 68
 Server Version: 19.03.5
 Storage Driver: overlay2
  Backing Filesystem: xfs
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: b34a5c8af56e510852c35414db4c1f4fa6172339
 runc version: 3e425f80a8c931f88e6d94a8c831b9d5aa481657
 init version: fec3683
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 5.1.12-300.fc30.x86_64
 Operating System: Fedora 30 (Server Edition)
 OSType: linux
 Architecture: x86_64
 CPUs: 4
 Total Memory: 25.5GiB
 Name: fedora1
 ID: ABPS:R6F2:FB6G:QKPG:IH26:LSEW:CHKC:7S7I:IGUA:N46P:NHRD:WMJN
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, Docker for MAC, physical, etc.):**
Physical server with Fedora

**Comment by vlk-charles** [comment]
I originally thought this was a 3.10 vs 3.9 issue but I was able to bisect it to a specific release. The bug was introduced in 3.10.0-debian-10-r45. I slightly edited the original report accordingly.

I think I found [the offending change](https://github.com/bitnami/bitnami-docker-moodle/commit/fd23323227d6a9801a06c974abe34d4036b4db47) in that release. The `moodle_upgrade` function in `libmoodle.sh` was changed. Near the end, the variable `moodle_install_args` is referenced but I think this should be `moodle_upgrade_args`.
